pkg/drivers/krunkit/krunkit.go

```diff
@@ -0,0 +1,562 @@
+		return nil
+	}
+	log.Debugf("Kill krunkit (pid=%d)", pid)
+	if err := process.Kill(pid, driverName); err != nil {
+		if err != os.ErrProcessDone {
+			return err
+		}
+		log.Debugf("Remove krunkit pidfile %q", pidfile)
+		// No process, stale pidfile.
+		if err := os.Remove(pidfile); err != nil {
+			log.Debugf("failed to remove %q: %s", pidfile, err)
+		}
+	}
+	return nil
+}
+
+func (d *Driver) publicSSHKeyPath() string {
+	return d.GetSSHKeyPath() + ".pub"
+}
+
+func (d *Driver) diskPath() string {
+	return d.ResolveStorePath("disk.img")
+}
+
+func (d *Driver) pidfilePath() string {
+	return d.ResolveStorePath(pidFileName)
+}
+
+func (d *Driver) logfilePath() string {
+	return d.ResolveStorePath(logFileName)
+}
+
+func (d *Driver) serialPath() string {
+	return d.ResolveStorePath(serialFileName)
+}
+
+func (d *Driver) isoPath() string {
+	return d.ResolveStorePath(isoFileName)
+}
+
+func (d *Driver) sockfilePath() string {
+	return d.ResolveStorePath(sockFileName)
+}
+
+func (d *Driver) restfulURI() string {
+	return fmt.Sprintf("unix://%s", d.sockfilePath())
+}
+
+func (d *Driver) vmStateURI() string {
+	return "http://_/vm/state"
+}
+
+// Make a boot2docker VM disk image.
+func (d *Driver) generateDiskImage(size int) error {
+	log.Debugf("Creating %d MB hard disk image...", size)
+
+	magicString := "boot2docker, please format-me"
+
+	buf := new(bytes.Buffer)
+	tw := tar.NewWriter(buf)
+
+	// magicString first so the automount script knows to format the disk
+	file := &tar.Header{Name: magicString, Size: int64(len(magicString))}
+	if err := tw.WriteHeader(file); err != nil {
+		return err
+	}
+	if _, err := tw.Write([]byte(magicString)); err != nil {
+		return err
+	}
+	// .ssh/key.pub => authorized_keys
+	file = &tar.Header{Name: ".ssh", Typeflag: tar.TypeDir, Mode: 0700}
+	if err := tw.WriteHeader(file); err != nil {
+		return err
+	}
+	pubKey, err := os.ReadFile(d.publicSSHKeyPath())
+	if err != nil {
+		return err
+	}
+	file = &tar.Header{Name: ".ssh/authorized_keys", Size: int64(len(pubKey)), Mode: 0644}
+	if err := tw.WriteHeader(file); err != nil {
+		return err
+	}
+	if _, err := tw.Write(pubKey); err != nil {
+		return err
+	}
+	file = &tar.Header{Name: ".ssh/authorized_keys2", Size: int64(len(pubKey)), Mode: 0644}
+	if err := tw.WriteHeader(file); err != nil {
+		return err
+	}
+	if _, err := tw.Write(pubKey); err != nil {
+		return err
+	}
+	if err := tw.Close(); err != nil {
+		return err
+	}
+	rawFile := d.diskPath()
+	if err := os.WriteFile(rawFile, buf.Bytes(), 0644); err != nil {
+		return nil
+	}
+	if err := os.Truncate(rawFile, int64(size)*int64(1024*1024)); err != nil {
+		return nil
+	}
+	log.Debugf("DONE writing to %s and %s", rawFile, d.diskPath())
+	return nil
+}
+
+func httpUnixClient(path string) http.Client {
+	return http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", path)
+			},
+		},
+	}
+}
+
+type VMState struct {
```

Comment:
does it need to be exported ? if yes comment

Reply:
Delete, not needed when using `--restful-url none://`

Reply:
Renamed